src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1105,14 +1105,6 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                 throw new BadImageFormatException();
             }
 
-            // This block enforces the rule that methods with [UnmanagedCallersOnly] attribute
-            // can only be called from unmanaged code. The call from managed code is replaced
-            // with a stub that throws an InvalidProgramException
-            if (method.IsUnmanagedCallersOnly && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0)
-            {
-                ThrowHelper.ThrowInvalidProgramException(ExceptionStringID.InvalidProgramUnmanagedCallersOnly, method);
```

Comment:
What will happen if someone tries to call UnmanagedCallersOnly method from managed code after this?

Reply:
I have pushed PR to print same error message as the JIT-based version of the runtime.